### PR TITLE
clusterversion: clean up old gates

### PR DESCRIPTION
**Pull request by RaduBerinde**
Epic: REL-1163
Release note: None

### CODE REVIEW

**Review by arulajmani — APPROVED** [review]
> Reviewed 8 of 28 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128820)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jbowens, @RaduBerinde, and @stevendanna)

___
*[`pkg/kv/kvserver/replica_range_lease.go` line 743 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128820#-O47_lIe1RKlDw_ePyHw:-O47_lIe1RKlDw_ePyHx:b-gwih5v) ([raw file](https://github.com/cockroachdb/cockroach/blob/fae1131227d619ac2721d266a6e9e51a69d0f9ed/pkg/kv/kvserver/replica_range_lease.go#L743)):*
> ```Go
> 		DisableAboveRaftLeaseTransferSafetyChecks: r.store.cfg.TestingKnobs.DisableAboveRaftLeaseTransferSafetyChecks,
> 		AllowLeaseProposalWhenNotLeader:           r.store.cfg.TestingKnobs.AllowLeaseRequestProposalsWhenNotLeader,
> 		ExpToEpochEquiv:                           true,
> ```

Would you mind adding a TODO to remove this fields from `leases.Settings` entirely? You can put my name on it.

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128820)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @jbowens, and @stevendanna)

___
*[`pkg/kv/kvserver/replica_range_lease.go` line 743 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128820#-O47_lIe1RKlDw_ePyHw:-O47dID39cIFRTzhyL-w:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fae1131227d619ac2721d266a6e9e51a69d0f9ed/pkg/kv/kvserver/replica_range_lease.go#L743)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Would you mind adding a TODO to remove this fields from `leases.Settings` entirely? You can put my name on it.
</blockquote></details>

Done.
___
*[`pkg/sql/catalog/lease/count.go` line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/128820#-O47dLhWBBJsyAnQmMWg:-O47dLhWBBJsyAnQmMWh:b4zf5aa) ([raw file](https://github.com/cockroachdb/cockroach/blob/fae1131227d619ac2721d266a6e9e51a69d0f9ed/pkg/sql/catalog/lease/count.go#L87)):*
> ```Go
> 		systemDBVersion = &v
> 	}
> 	// TODO(radu): this used to check the version against a pre-v24.1 gate; should
> ```

@arulajmani - should I just assume this is always `true` and remove it? I don't know if `systemDBVersion == nil` is a real case.

File: pkg/ccl/backupccl/restore_job.go [hunk 1]
```diff
@@ -317,7 +317,8 @@ func restore(
 		return emptyRowCount, err
 	}
 
-	on231 := clusterversion.TODO_Delete_V23_1.Version().LessEq(job.Payload().CreationClusterVersion)
+	ver := job.Payload().CreationClusterVersion
+	on231 := ver.Major > 23 || (ver.Major == 23 && ver.Minor >= 1)
```
Comment (stevendanna) [inline pkg/ccl/backupccl/restore_job.go:321]: I _think_ that with MinVersion set to 24_1 the only way we were get here is if the user has started their restore with `unsafe_restore_incompatible_version`.   Doesn't need to happen in this PR, but seems like we might be able to just remove this now. cc @msbutler

**Review by stevendanna — APPROVED** [review]
Just reviewed the backupccl code.

File: pkg/ccl/backupccl/restore_job.go (hunk 1 above)
Comment (RaduBerinde) [reply pkg/ccl/backupccl/restore_job.go]: Adding just a TODO in this PR.

**Pull request merged**